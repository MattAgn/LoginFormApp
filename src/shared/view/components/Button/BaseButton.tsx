import styled from '@emotion/native';
import React from 'react';
import { Pressable, TextStyle } from 'react-native';

import { ButtonProps } from '#shared/view/components/Button/Button';
import { Loader } from '#shared/view/components/Loader/Loader';
import { Typography } from '#shared/view/components/Typography/Typography';
import { BaseButtonStyle, ButtonState } from '#shared/view/theme/button';
import { ThemeColor } from '#shared/view/theme/colors.types';

const BUTTON_ICON_SIZE_PX = 24;

type BaseButtonProps = ButtonProps & {
  style: BaseButtonStyle;
};

const BaseButtonUnmemoized = ({
  isDisabled = false,
  isLoading,
  onPress,
  label,
  StartIcon,
  EndIcon,
  style,
  accessibilityLabel,
}: BaseButtonProps) => {
  // https://www.bam.tech/article/how-insidious-reactnative-onpress-could-be
  const handlePress = () => onPress();

  return (
    <Pressable
      onPress={handlePress}
      disabled={isLoading || isDisabled}
      style={({ pressed }) =>
        style[getButtonState({ isDisabled, isPressed: pressed })].container
      }
      accessibilityLabel={accessibilityLabel ?? label}
      accessibilityRole="button"
      accessibilityValue={isLoading ? { text: 'Loading' } : undefined}
    >
      {({ pressed }) => (
        <ContentContainer>
          <ButtonContent
            StartIcon={StartIcon}
            EndIcon={EndIcon}
            isLoading={isLoading}
            label={label}
            textStyle={
              style[getButtonState({ isDisabled, isPressed: pressed })].text
            }
          />
        </ContentContainer>
      )}
    </Pressable>
  );
};

export const BaseButton = React.memo(BaseButtonUnmemoized);

type ButtonContentProps = Pick<
  BaseButtonProps,
  'EndIcon' | 'StartIcon' | 'isLoading' | 'label'
> & { textStyle: { color: ThemeColor } & TextStyle };

const ButtonContent = ({
  StartIcon,
  EndIcon,
  isLoading,
  label,
  textStyle,
}: ButtonContentProps) => {
  if (isLoading) {
    return (
      <Loader
        size={BUTTON_ICON_SIZE_PX}
        color={textStyle.color}
        testID="activity-indicator"
      />
    );
  }

  return (
    <>
      {StartIcon ? (
        <StartIcon color={textStyle.color} size={BUTTON_ICON_SIZE_PX} />
      ) : null}
      <Typography.P1Bold style={textStyle}>{label}</Typography.P1Bold>
      {EndIcon ? (
        <EndIcon color={textStyle.color} size={BUTTON_ICON_SIZE_PX} />
      ) : null}
    </>
  );
};

type GetButtonStateParams = {
  isDisabled: boolean;
  isPressed: boolean;
};

const getButtonState = ({
  isDisabled,
  isPressed,
}: GetButtonStateParams): ButtonState => {
  if (isDisabled) {
    return 'disabled';
  }

  if (isPressed) {
    return 'active';
  }

  return 'rest';
};

const ContentContainer = styled.View(() => ({
  alignItems: 'center',
  flexDirection: 'row',
  justifyContent: 'center',
  gap: 8,
}));
