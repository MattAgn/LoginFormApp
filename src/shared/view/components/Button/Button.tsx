import { useTheme } from '@emotion/react';

import { BaseButton } from '#shared/view/components/Button/BaseButton';
import { IconType } from '#shared/view/icons/Icon.types';

export type ButtonProps = {
  isDisabled?: boolean;
  isLoading?: boolean;
  onPress: () => void;
  label: string;
  StartIcon?: IconType;
  EndIcon?: IconType;
  accessibilityLabel?: string;
};

export const Button = {
  Primary: (props: ButtonProps) => {
    const theme = useTheme();
    return <BaseButton style={theme.button.primary} {...props} />;
  },
  Secondary: (props: ButtonProps) => {
    const theme = useTheme();
    return <BaseButton style={theme.button.secondary} {...props} />;
  },
  Tertiary: (props: ButtonProps) => {
    const theme = useTheme();
    return <BaseButton style={theme.button.tertiary} {...props} />;
  },
};
