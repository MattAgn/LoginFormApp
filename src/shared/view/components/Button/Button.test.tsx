import { act, screen, userEvent } from '@testing-library/react-native';

import { renderWithProviders } from '#testing/render';

import { CrossIcon } from '../../icons/Cross.icon';
import { Button } from './Button';

const isLoading = [true, false];
const isDisabled = [true, false];

const differentTestCases = isLoading.flatMap((loadingElement) =>
  isDisabled.map((disabledElement) => ({
    isLoading: loadingElement,
    isDisabled: disabledElement,
  })),
);

describe('Button Component', () => {
  it('renders the label', () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <Button.Primary label="Button primary" onPress={mockOnPress} />,
    );

    expect(screen.getByText('Button primary')).toBeOnTheScreen();
  });

  it('renders the icon', () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <Button.Primary
        label="Button primary"
        onPress={mockOnPress}
        StartIcon={CrossIcon}
      />,
    );

    expect(screen.getByTestId('cross-icon')).toBeOnTheScreen();
  });

  it('calls the onPress', async () => {
    const mockOnPress = jest.fn();

    renderWithProviders(<Button.Primary label="test" onPress={mockOnPress} />);

    await userEvent.press(screen.getByText('test'));

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it(`does not call onPress when button is pressed and disabled is true`, async () => {
    const mockOnPress = jest.fn();

    renderWithProviders(
      <Button.Primary label="test" onPress={mockOnPress} isDisabled={true} />,
    );

    await userEvent.press(screen.getByText('test'));

    expect(mockOnPress).toHaveBeenCalledTimes(0);
  });

  it.each(differentTestCases)(
    'renders Primary Button with props %p',
    (props) => {
      renderWithProviders(
        <Button.Primary
          label="Primary button"
          onPress={jest.fn()}
          {...props}
        />,
      );

      expect(screen).toMatchComponentSnapshot();
    },
  );

  it.each(differentTestCases)(
    'renders Secondary Button with props %p',
    (props) => {
      renderWithProviders(
        <Button.Secondary
          label="Secondary button"
          onPress={jest.fn()}
          {...props}
        />,
      );

      expect(screen).toMatchComponentSnapshot();
    },
  );

  it.each(differentTestCases)(
    'renders Tertiary Button with props %p',
    (props) => {
      renderWithProviders(
        <Button.Tertiary
          label="Tertiary button"
          onPress={jest.fn()}
          {...props}
        />,
      );

      expect(screen).toMatchComponentSnapshot();
    },
  );

  test('primary button pressed matches snapshot', async () => {
    renderWithProviders(
      <Button.Primary label="Primary button" onPress={jest.fn()} />,
    );

    const button = screen.getByRole('button', { name: 'Primary button' });

    await act(async () => {
      button.props.onResponderGrant({
        persist: jest.fn(),
        currentTarget: { measure: jest.fn() },
        nativeEvent: {
          changedTouches: [],
          identifier: 0,
          locationX: 0,
          locationY: 0,
          pageX: 0,
          pageY: 0,
          target: 0,
          timestamp: Date.now(),
          touches: [],
        },
        dispatchConfig: { registrationName: 'onResponderGrant' },
      });
    });

    expect(screen).toMatchComponentSnapshot();
  });

  test('secondary button pressed matches snapshot', async () => {
    renderWithProviders(
      <Button.Secondary label="Secondary button" onPress={jest.fn()} />,
    );

    const button = screen.getByRole('button', { name: 'Secondary button' });

    await act(async () => {
      button.props.onResponderGrant({
        persist: jest.fn(),
        currentTarget: { measure: jest.fn() },
        nativeEvent: {
          changedTouches: [],
          identifier: 0,
          locationX: 0,
          locationY: 0,
          pageX: 0,
          pageY: 0,
          target: 0,
          timestamp: Date.now(),
          touches: [],
        },
        dispatchConfig: { registrationName: 'onResponderGrant' },
      });
    });

    expect(screen).toMatchComponentSnapshot();
  });

  test('tertiary button pressed matches snapshot', async () => {
    renderWithProviders(
      <Button.Tertiary label="Tertiary button" onPress={jest.fn()} />,
    );

    const button = screen.getByRole('button', { name: 'Tertiary button' });

    await act(async () => {
      button.props.onResponderGrant({
        persist: jest.fn(),
        currentTarget: { measure: jest.fn() },
        nativeEvent: {
          changedTouches: [],
          identifier: 0,
          locationX: 0,
          locationY: 0,
          pageX: 0,
          pageY: 0,
          target: 0,
          timestamp: Date.now(),
          touches: [],
        },
        dispatchConfig: { registrationName: 'onResponderGrant' },
      });
    });

    expect(screen).toMatchComponentSnapshot();
  });
});
