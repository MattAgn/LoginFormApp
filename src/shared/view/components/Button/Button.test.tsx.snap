// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component primary button pressed matches snapshot 1`] = `"Primary button"`;

exports[`Button Component primary button pressed matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Primary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF7E1D",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 0.5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Primary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Primary Button with props {"isDisabled": false, "isLoading": false} 1`] = `"Primary button"`;

exports[`Button Component renders Primary Button with props {"isDisabled": false, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Primary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF7E1D",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Primary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Primary Button with props {"isDisabled": false, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Primary Button with props {"isDisabled": false, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Primary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF7E1D",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#FFFFFF"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Primary Button with props {"isDisabled": true, "isLoading": false} 1`] = `"Primary button"`;

exports[`Button Component renders Primary Button with props {"isDisabled": true, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Primary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#737373",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Primary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Primary Button with props {"isDisabled": true, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Primary Button with props {"isDisabled": true, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Primary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#737373"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Secondary Button with props {"isDisabled": false, "isLoading": false} 1`] = `"Secondary button"`;

exports[`Button Component renders Secondary Button with props {"isDisabled": false, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Secondary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#FF7E1D",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FF7E1D",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Secondary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Secondary Button with props {"isDisabled": false, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Secondary Button with props {"isDisabled": false, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Secondary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#FF7E1D",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#FF7E1D"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Secondary Button with props {"isDisabled": true, "isLoading": false} 1`] = `"Secondary button"`;

exports[`Button Component renders Secondary Button with props {"isDisabled": true, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Secondary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderColor": "#737373",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#737373",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Secondary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Secondary Button with props {"isDisabled": true, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Secondary Button with props {"isDisabled": true, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Secondary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "borderColor": "#737373",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#737373"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": false, "isLoading": false} 1`] = `"Tertiary button"`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": false, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Tertiary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FF7E1D",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "underline",
          }
        }
      >
        Tertiary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": false, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": false, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Tertiary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#FF7E1D"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": true, "isLoading": false} 1`] = `"Tertiary button"`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": true, "isLoading": false} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Tertiary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#737373",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "underline",
          }
        }
      >
        Tertiary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": true, "isLoading": true} 1`] = `null`;

exports[`Button Component renders Tertiary Button with props {"isDisabled": true, "isLoading": true} 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Tertiary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": "Loading",
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <View
        size={24}
        style={
          {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <ActivityIndicator
          color="#737373"
          testID="activity-indicator"
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component secondary button pressed matches snapshot 1`] = `"Secondary button"`;

exports[`Button Component secondary button pressed matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Secondary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#FF7E1D",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 0.5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FF7E1D",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "none",
          }
        }
      >
        Secondary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button Component tertiary button pressed matches snapshot 1`] = `"Tertiary button"`;

exports[`Button Component tertiary button pressed matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="Tertiary button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 40,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 0.5,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={0}
        style={
          {
            "color": "#FF7E1D",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "lineHeight": 24,
            "textDecorationLine": "underline",
          }
        }
      >
        Tertiary button
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
